includes/import/abstract-wc-product-importer.php

```diff
@@ -0,0 +1,941 @@
+			}
+
+			$classname = WC_Product_Factory::get_classname_from_product_type( $data['type'] );
+
+			if ( ! class_exists( $classname ) ) {
+				$classname = 'WC_Product_Simple';
+			}
+
+			$product = new $classname( $id );
+		} elseif ( isset( $data['id'] ) ) {
+			$product = wc_get_product( $id );
+		} else {
+			$product = new WC_Product_Simple();
+		}
+
+		// @todo need to check first how we'll handle variations.
+		if ( 'variation' === $product->get_type() ) {
+			$product = $this->save_variation_data( $product, $data );
+		} else {
+			$product = $this->save_product_data( $product, $data );
+		}
+
+		return apply_filters( 'woocommerce_product_csv_import_pre_insert_product_object', $product, $data );
+	}
+
+	/**
+	 * Set product data.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param array      $data    Row data.
+	 *
+	 * @return WC_Product
+	 */
+	protected function save_product_data( $product, $data ) {
+
+		// Post title.
+		if ( isset( $data['name'] ) ) {
+			$product->set_name( wp_filter_post_kses( $data['name'] ) );
+		}
+
+		// Post content.
+		if ( isset( $data['description'] ) ) {
+			$product->set_description( wp_filter_post_kses( $data['description'] ) );
+		}
+
+		// Post excerpt.
+		if ( isset( $data['short_description'] ) ) {
+			$product->set_short_description( wp_filter_post_kses( $data['short_description'] ) );
+		}
+
+		// Post status.
+		if ( isset( $data['status'] ) ) {
+			$product->set_status( $data['status'] ? 'publish' : 'draft' );
+		}
+
+		// Post slug.
+		if ( isset( $data['slug'] ) ) {
+			$product->set_slug( $data['slug'] );
+		}
+
+		// Menu order.
+		if ( isset( $data['menu_order'] ) ) {
+			$product->set_menu_order( $data['menu_order'] );
+		}
+
+		// Comment status.
+		if ( isset( $data['reviews_allowed'] ) ) {
+			$product->set_reviews_allowed( $data['reviews_allowed'] );
+		}
+
+		// Virtual.
+		if ( isset( $data['virtual'] ) ) {
+			$product->set_virtual( $data['virtual'] );
+		}
+
+		// Tax status.
+		if ( isset( $data['tax_status'] ) ) {
+			$product->set_tax_status( $data['tax_status'] );
+		}
+
+		// Tax Class.
+		if ( isset( $data['tax_class'] ) ) {
+			$product->set_tax_class( $data['tax_class'] );
+		}
+
+		// Catalog Visibility.
+		if ( isset( $data['catalog_visibility'] ) ) {
+			$product->set_catalog_visibility( $data['catalog_visibility'] );
+		}
+
+		// Purchase Note.
+		if ( isset( $data['purchase_note'] ) ) {
+			$product->set_purchase_note( wc_clean( $data['purchase_note'] ) );
+		}
+
+		// Featured Product.
+		if ( isset( $data['featured'] ) ) {
+			$product->set_featured( $data['featured'] );
+		}
+
+		// Shipping data.
+		$product = $this->save_product_shipping_data( $product, $data );
+
+		// SKU.
+		if ( isset( $data['sku'] ) ) {
+			$product->set_sku( wc_clean( $data['sku'] ) );
+		}
+
+		// Attributes.
+		if ( isset( $data['attributes'] ) ) {
```

Comment:
Should this logic move to the parser? So the data passed is formatted first. cc @claudiulodro